torch/nn/functional.py

```diff
@@ -458,10 +458,14 @@ def fractional_max_pool2d_with_indices(
     if output_size is None:
         assert output_ratio is not None
         _output_ratio = _pair(output_ratio)
-        output_size = [int(input.size(2) * _output_ratio[0]), int(input.size(3) * _output_ratio[1])]
+        output_size = [int(input.size(-2) * _output_ratio[0]), int(input.size(-1) * _output_ratio[1])]
 
     if _random_samples is None:
-        _random_samples = torch.rand(input.size(0), input.size(1), 2, dtype=input.dtype, device=input.device)
+        n_batch = input.size(0)
+        if input.dim() == 3:
+            # no_batch dimension
+            n_batch = 1
+        _random_samples = torch.rand(n_batch, input.size(-3), 2, dtype=input.dtype, device=input.device)
```

Comment:
I'm open to making `_random_samples.dim() == 3`.

Reply:
Probably the way to go for the cleanest solution